### ISSUE TITLE
BUG: building current SciPy main with MSVC leads to broken install

### ISSUE THREAD

**Issue by h-vetinari** [scipy/scipy#21561]
While testing https://github.com/scipy/scipy/pull/21553 in https://github.com/conda-forge/scipy-feedstock/pull/284, we ran into issues with the windows builds passing compilation, but then failing in some hard-to-diagnose way when importing modules.

This is obviously a question about the lack of MSVC-based CI; I'm planning to add this as soon as LLVM 19 goes GA (and we have built the stack in conda-forge). In the meantime, we might have to bisect to figure out what broke.

**Comment by ilayn** [comment]
Just checking recent merged PRs say https://github.com/scipy/scipy/actions/runs/9643479376/job/26593572266 it looks like MSVC ifx openblas job went through.

**Comment by h-vetinari** [comment]
```
git bisect start  # on current main, 1eff03d65543311c392c9f137ed490fd3c322c00
git bisect bad
git bisect good tags/v1.14.0
git bisect good c9ae6f6759947df245e8b6319cf9b548f3340032  # ignored
git bisect bad f406ec58a51fed670a3cc83788184340f6f620b6
git bisect good 485b26da6b797b5b98d9441ae304388ea1dceb66
git bisect ??? a839e298a405b0b8d57ed5e025d132edd42f43ef
```

**Comment by lucascolley** [comment]
the error message from conda-build really doesn't look helpful :sweat_smile:

**Comment by rgommers** [comment]
In case this becomes a problem, I'd suggest reverting gh-21225 (since that change doesn't really matter either way) rather than burning a lot of time on this.

Side note: actual MSVC doesn't work with Pythran; we need `clang-cl` instead.

**Comment by h-vetinari** [comment]
You're right, it's clang-cl. I was planning to add a job here that builds +/- like the feedstock and see if the problem persist on main as well. If that doesn't give any indication (or we're getting close to the release and there's still no information), then I'm in favour of reverting as well

**Comment by lucascolley** [comment]
we have CI for conda-forge clang-cl builds in the Pixi packages job now, closing
